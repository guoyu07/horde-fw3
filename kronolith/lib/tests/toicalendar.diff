001- BEGIN:VCALENDAR
001+ Warning: include_once(/home/wrobel/usr/devel/gh/horde-glue/horde-fw3/kronolith/lib/Driver/.php): failed to open stream: No such file or directory in /home/wrobel/usr/devel/gh/horde-glue/horde-fw3/kronolith/lib/Driver.php on line 198
002- VERSION:1.0
002+ 
003- PRODID:-//The Horde Project//Horde_iCalendar Library//EN
003+ Warning: include_once(): Failed opening '/home/wrobel/usr/devel/gh/horde-glue/horde-fw3/kronolith/lib/Driver/.php' for inclusion (include_path='.:/usr/share/php5:/usr/share/php') in /home/wrobel/usr/devel/gh/horde-glue/horde-fw3/kronolith/lib/Driver.php on line 198
004- METHOD:PUBLISH
004+ 
005- BEGIN:VEVENT
005+ A fatal error has occurred
006- DTSTART:20070315T121020Z
006+ 
007- DTEND:20070315T132000Z
007+ You must configure a DataTree backend to use Shares.
008- DTSTAMP:%d%d%d%d%d%d%d%dT%d%d%d%d%d%dZ
008+ 
009- UID:20070315143732.4wlenqz3edq8@horde.org
009+ Details have been logged for the administrator.
010- SUMMARY:Event from 1:10pm to 2:20pm
011- ORGANIZER;CN=joe:mailto:joe
012- DESCRIPTION;ENCODING=QUOTED-PRINTABLE;CHARSET=ISO-8859-1:Sch=F6ne Bescherung=0D=0A=
013- New line
014- CATEGORIES;ENCODING=QUOTED-PRINTABLE;CHARSET=ISO-8859-1:Sch=F6ngeistiges
015- LOCATION;ENCODING=QUOTED-PRINTABLE;CHARSET=ISO-8859-1:Allg=E4u
016- CLASS:PUBLIC
017- STATUS:CONFIRMED
018- TRANSP:0
019- AALARM:20070315T120020Z
020- RRULE:D2 #0
021- EXDATE:20070319T121020Z
022- END:VEVENT
023- END:VCALENDAR
024- 
025- BEGIN:VCALENDAR
026- VERSION:2.0
027- PRODID:-//The Horde Project//Horde_iCalendar Library//EN
028- METHOD:PUBLISH
029- BEGIN:VEVENT
030- DTSTART:20070315T121020Z
031- DTEND:20070315T132000Z
032- DTSTAMP:%d%d%d%d%d%d%d%dT%d%d%d%d%d%dZ
033- UID:20070315143732.4wlenqz3edq8@horde.org
034- SUMMARY:Event from 1:10pm to 2:20pm
035- ORGANIZER;CN=joe:mailto:joe
036- DESCRIPTION:Schöne Bescherung\nNew line
037- CATEGORIES:Schöngeistiges
038- LOCATION:Allgäu
039- CLASS:PUBLIC
040- STATUS:CONFIRMED
041- TRANSP:OPAQUE
042- RRULE:FREQ=DAILY;INTERVAL=2
043- EXDATE:20070319T121020Z
044- BEGIN:VALARM
045- ACTION:DISPLAY
046- TRIGGER;VALUE=DURATION:-PT10M
047- END:VALARM
048- END:VEVENT
049- END:VCALENDAR
050- 
051- BEGIN:VCALENDAR
052- VERSION:1.0
053- PRODID:-//The Horde Project//Horde_iCalendar Library//EN
054- METHOD:PUBLISH
055- BEGIN:VEVENT
056- DTSTART:20070315T121020Z
057- DTEND:20070315T132000Z
058- DTSTAMP:%d%d%d%d%d%d%d%dT%d%d%d%d%d%dZ
059- UID:20070315143732.4wlenqz3edq8@horde.org
060- SUMMARY:Private Event from 1:10pm to 2:20pm
061- ORGANIZER;CN=joe:mailto:joe
062- CLASS:PRIVATE
063- STATUS:TENTATIVE
064- TRANSP:0
065- ATTENDEE;EXPECT=REQUIRE;STATUS=NEEDS ACTION;RSVP=YES;ENCODING=QUOTED-PRINTABLE;CHARSET=ISO-8859-1:J=FCrgen Doe <juergen@example.com>
066- ATTENDEE;EXPECT=REQUEST;STATUS=ACCEPTED:Jane Doe
067- ATTENDEE;EXPECT=FYI;STATUS=DECLINED:Jack Doe <jack@example.com>
068- ATTENDEE;EXPECT=FYI;STATUS=TENTATIVE:jenny@example.com
069- AALARM:20070315T120020Z
070- RRULE:MD1 15 #0
071- EXDATE:20070415T111020Z
072- END:VEVENT
073- END:VCALENDAR
074- 
075- BEGIN:VCALENDAR
076- VERSION:2.0
077- PRODID:-//The Horde Project//Horde_iCalendar Library//EN
078- METHOD:PUBLISH
079- BEGIN:VEVENT
080- DTSTART:20070315T121020Z
081- DTEND:20070315T132000Z
082- DTSTAMP:%d%d%d%d%d%d%d%dT%d%d%d%d%d%dZ
083- UID:20070315143732.4wlenqz3edq8@horde.org
084- SUMMARY:Private Event from 1:10pm to 2:20pm
085- ORGANIZER;CN=joe:mailto:joe
086- CLASS:PRIVATE
087- STATUS:TENTATIVE
088- TRANSP:OPAQUE
089- ATTENDEE;ROLE=REQ-PARTICIPANT;PARTSTAT=NEEDS-ACTION;RSVP=TRUE;CN=Jürgen
090-   Doe:mailto:juergen@example.com
091- ATTENDEE;ROLE=OPT-PARTICIPANT;PARTSTAT=ACCEPTED;CN=Jane Doe:
092- ATTENDEE;ROLE=NON-PARTICIPANT;PARTSTAT=DECLINED;CN=Jack
093-   Doe:mailto:jack@example.com
094- ATTENDEE;ROLE=NON-PARTICIPANT;PARTSTAT=TENTATIVE:mailto:jenny@example.com
095- RRULE:FREQ=MONTHLY;INTERVAL=1
096- EXDATE:20070415T111020Z
097- BEGIN:VALARM
098- ACTION:DISPLAY
099- TRIGGER;VALUE=DURATION:-PT10M
100- END:VALARM
101- END:VEVENT
102- END:VCALENDAR