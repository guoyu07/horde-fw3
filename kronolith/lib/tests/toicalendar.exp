BEGIN:VCALENDAR
VERSION:1.0
PRODID:-//The Horde Project//Horde_iCalendar Library, Horde 3.3.4-cvs//EN
METHOD:PUBLISH
BEGIN:VEVENT
DTSTART:20070315T121020Z
DTEND:20070315T132000Z
DTSTAMP:%d%d%d%d%d%d%d%dT%d%d%d%d%d%dZ
UID:20070315143732.4wlenqz3edq8@horde.org
SUMMARY:Event from 1:10pm to 2:20pm
ORGANIZER;CN=joe:mailto:joe
DESCRIPTION;ENCODING=QUOTED-PRINTABLE;CHARSET=ISO-8859-1:Sch=F6ne Bescherung=0D=0A=
New line
CATEGORIES;ENCODING=QUOTED-PRINTABLE;CHARSET=ISO-8859-1:Sch=F6ngeistiges
LOCATION;ENCODING=QUOTED-PRINTABLE;CHARSET=ISO-8859-1:Allg=E4u
CLASS:PUBLIC
STATUS:CONFIRMED
TRANSP:0
AALARM:20070315T120020Z
RRULE:D2 #0
EXDATE:20070319T121020Z
END:VEVENT
END:VCALENDAR

BEGIN:VCALENDAR
VERSION:2.0
PRODID:-//The Horde Project//Horde_iCalendar Library\, Horde 3.3.4-cvs//EN
METHOD:PUBLISH
BEGIN:VEVENT
DTSTART:20070315T121020Z
DTEND:20070315T132000Z
DTSTAMP:%d%d%d%d%d%d%d%dT%d%d%d%d%d%dZ
UID:20070315143732.4wlenqz3edq8@horde.org
SUMMARY:Event from 1:10pm to 2:20pm
ORGANIZER;CN=joe:mailto:joe
DESCRIPTION:Schöne Bescherung\nNew line
CATEGORIES:Schöngeistiges
LOCATION:Allgäu
CLASS:PUBLIC
STATUS:CONFIRMED
TRANSP:OPAQUE
RRULE:FREQ=DAILY;INTERVAL=2
EXDATE:20070319T121020Z
BEGIN:VALARM
ACTION:DISPLAY
TRIGGER;VALUE=DURATION:-PT10M
END:VALARM
END:VEVENT
END:VCALENDAR

BEGIN:VCALENDAR
VERSION:1.0
PRODID:-//The Horde Project//Horde_iCalendar Library, Horde 3.3.4-cvs//EN
METHOD:PUBLISH
BEGIN:VEVENT
DTSTART:20070315T121020Z
DTEND:20070315T132000Z
DTSTAMP:%d%d%d%d%d%d%d%dT%d%d%d%d%d%dZ
UID:20070315143732.4wlenqz3edq8@horde.org
SUMMARY:Private Event from 1:10pm to 2:20pm
ORGANIZER;CN=joe:mailto:joe
CLASS:PRIVATE
STATUS:TENTATIVE
TRANSP:0
ATTENDEE;EXPECT=REQUIRE;STATUS=NEEDS ACTION;RSVP=YES;ENCODING=QUOTED-PRINTABLE;CHARSET=ISO-8859-1:J=FCrgen Doe <juergen@example.com>
ATTENDEE;EXPECT=REQUEST;STATUS=ACCEPTED:Jane Doe
ATTENDEE;EXPECT=FYI;STATUS=DECLINED:Jack Doe <jack@example.com>
ATTENDEE;EXPECT=FYI;STATUS=TENTATIVE:jenny@example.com
AALARM:20070315T120020Z
RRULE:MD1 15 #0
EXDATE:20070415T111020Z
END:VEVENT
END:VCALENDAR

BEGIN:VCALENDAR
VERSION:2.0
PRODID:-//The Horde Project//Horde_iCalendar Library\, Horde 3.3.4-cvs//EN
METHOD:PUBLISH
BEGIN:VEVENT
DTSTART:20070315T121020Z
DTEND:20070315T132000Z
DTSTAMP:%d%d%d%d%d%d%d%dT%d%d%d%d%d%dZ
UID:20070315143732.4wlenqz3edq8@horde.org
SUMMARY:Private Event from 1:10pm to 2:20pm
ORGANIZER;CN=joe:mailto:joe
CLASS:PRIVATE
STATUS:TENTATIVE
TRANSP:OPAQUE
ATTENDEE;ROLE=REQ-PARTICIPANT;PARTSTAT=NEEDS-ACTION;RSVP=TRUE;CN="Jürgen
  Doe":mailto:juergen@example.com
ATTENDEE;ROLE=OPT-PARTICIPANT;PARTSTAT=ACCEPTED;CN=Jane Doe:
ATTENDEE;ROLE=NON-PARTICIPANT;PARTSTAT=DECLINED;CN=Jack
  Doe:mailto:jack@example.com
ATTENDEE;ROLE=NON-PARTICIPANT;PARTSTAT=TENTATIVE:mailto:jenny@example.com
RRULE:FREQ=MONTHLY;INTERVAL=1
EXDATE:20070415T111020Z
BEGIN:VALARM
ACTION:DISPLAY
TRIGGER;VALUE=DURATION:-PT10M
END:VALARM
END:VEVENT
END:VCALENDAR