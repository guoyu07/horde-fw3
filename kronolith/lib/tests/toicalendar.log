
---- EXPECTED OUTPUT
BEGIN:VCALENDAR
VERSION:1.0
PRODID:-//The Horde Project//Horde_iCalendar Library//EN
METHOD:PUBLISH
BEGIN:VEVENT
DTSTART:20070315T121020Z
DTEND:20070315T132000Z
DTSTAMP:%d%d%d%d%d%d%d%dT%d%d%d%d%d%dZ
UID:20070315143732.4wlenqz3edq8@horde.org
SUMMARY:Event from 1:10pm to 2:20pm
ORGANIZER;CN=joe:mailto:joe
DESCRIPTION;ENCODING=QUOTED-PRINTABLE;CHARSET=ISO-8859-1:Sch=F6ne Bescherung=0D=0A=
New line
CATEGORIES;ENCODING=QUOTED-PRINTABLE;CHARSET=ISO-8859-1:Sch=F6ngeistiges
LOCATION;ENCODING=QUOTED-PRINTABLE;CHARSET=ISO-8859-1:Allg=E4u
CLASS:PUBLIC
STATUS:CONFIRMED
TRANSP:0
AALARM:20070315T120020Z
RRULE:D2 #0
EXDATE:20070319T121020Z
END:VEVENT
END:VCALENDAR

BEGIN:VCALENDAR
VERSION:2.0
PRODID:-//The Horde Project//Horde_iCalendar Library//EN
METHOD:PUBLISH
BEGIN:VEVENT
DTSTART:20070315T121020Z
DTEND:20070315T132000Z
DTSTAMP:%d%d%d%d%d%d%d%dT%d%d%d%d%d%dZ
UID:20070315143732.4wlenqz3edq8@horde.org
SUMMARY:Event from 1:10pm to 2:20pm
ORGANIZER;CN=joe:mailto:joe
DESCRIPTION:Schöne Bescherung\nNew line
CATEGORIES:Schöngeistiges
LOCATION:Allgäu
CLASS:PUBLIC
STATUS:CONFIRMED
TRANSP:OPAQUE
RRULE:FREQ=DAILY;INTERVAL=2
EXDATE:20070319T121020Z
BEGIN:VALARM
ACTION:DISPLAY
TRIGGER;VALUE=DURATION:-PT10M
END:VALARM
END:VEVENT
END:VCALENDAR

BEGIN:VCALENDAR
VERSION:1.0
PRODID:-//The Horde Project//Horde_iCalendar Library//EN
METHOD:PUBLISH
BEGIN:VEVENT
DTSTART:20070315T121020Z
DTEND:20070315T132000Z
DTSTAMP:%d%d%d%d%d%d%d%dT%d%d%d%d%d%dZ
UID:20070315143732.4wlenqz3edq8@horde.org
SUMMARY:Private Event from 1:10pm to 2:20pm
ORGANIZER;CN=joe:mailto:joe
CLASS:PRIVATE
STATUS:TENTATIVE
TRANSP:0
ATTENDEE;EXPECT=REQUIRE;STATUS=NEEDS ACTION;RSVP=YES;ENCODING=QUOTED-PRINTABLE;CHARSET=ISO-8859-1:J=FCrgen Doe <juergen@example.com>
ATTENDEE;EXPECT=REQUEST;STATUS=ACCEPTED:Jane Doe
ATTENDEE;EXPECT=FYI;STATUS=DECLINED:Jack Doe <jack@example.com>
ATTENDEE;EXPECT=FYI;STATUS=TENTATIVE:jenny@example.com
AALARM:20070315T120020Z
RRULE:MD1 15 #0
EXDATE:20070415T111020Z
END:VEVENT
END:VCALENDAR

BEGIN:VCALENDAR
VERSION:2.0
PRODID:-//The Horde Project//Horde_iCalendar Library//EN
METHOD:PUBLISH
BEGIN:VEVENT
DTSTART:20070315T121020Z
DTEND:20070315T132000Z
DTSTAMP:%d%d%d%d%d%d%d%dT%d%d%d%d%d%dZ
UID:20070315143732.4wlenqz3edq8@horde.org
SUMMARY:Private Event from 1:10pm to 2:20pm
ORGANIZER;CN=joe:mailto:joe
CLASS:PRIVATE
STATUS:TENTATIVE
TRANSP:OPAQUE
ATTENDEE;ROLE=REQ-PARTICIPANT;PARTSTAT=NEEDS-ACTION;RSVP=TRUE;CN=Jürgen
  Doe:mailto:juergen@example.com
ATTENDEE;ROLE=OPT-PARTICIPANT;PARTSTAT=ACCEPTED;CN=Jane Doe:
ATTENDEE;ROLE=NON-PARTICIPANT;PARTSTAT=DECLINED;CN=Jack
  Doe:mailto:jack@example.com
ATTENDEE;ROLE=NON-PARTICIPANT;PARTSTAT=TENTATIVE:mailto:jenny@example.com
RRULE:FREQ=MONTHLY;INTERVAL=1
EXDATE:20070415T111020Z
BEGIN:VALARM
ACTION:DISPLAY
TRIGGER;VALUE=DURATION:-PT10M
END:VALARM
END:VEVENT
END:VCALENDAR
---- ACTUAL OUTPUT
Warning: include_once(/home/wrobel/usr/devel/gh/horde-glue/horde-fw3/kronolith/lib/Driver/.php): failed to open stream: No such file or directory in /home/wrobel/usr/devel/gh/horde-glue/horde-fw3/kronolith/lib/Driver.php on line 198

Warning: include_once(): Failed opening '/home/wrobel/usr/devel/gh/horde-glue/horde-fw3/kronolith/lib/Driver/.php' for inclusion (include_path='.:/usr/share/php5:/usr/share/php') in /home/wrobel/usr/devel/gh/horde-glue/horde-fw3/kronolith/lib/Driver.php on line 198

A fatal error has occurred

You must configure a DataTree backend to use Shares.

Details have been logged for the administrator.
---- FAILED
