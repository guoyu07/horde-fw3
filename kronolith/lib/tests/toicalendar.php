<?php

require 'Horde/CLI.php';
Horde_CLI::init();
define('AUTH_HANDLER', true);
require dirname(__FILE__) . '/../base.php';
require 'Horde/iCalendar.php';

$driver = new Kronolith_Driver();
$object = new Kronolith_Event($driver);
$object->start = new Horde_Date('2007-03-15 13:10:20');
$object->end = new Horde_Date('2007-03-15 14:20:00');
$object->setCreatorId('joe');
$object->setUID('20070315143732.4wlenqz3edq8@horde.org');
$object->setTitle('Hübscher Termin');
$object->setDescription("Schöne Bescherung\nNew line");
$object->setCategory('Schöngeistiges');
$object->setLocation('Allgäu');
$object->setAlarm(10);
$object->recurrence = new Horde_Date_Recurrence($object->start);
$object->recurrence->setRecurType(HORDE_DATE_RECUR_DAILY);
$object->recurrence->setRecurInterval(2);
$object->recurrence->addException(2007, 3, 19);
$object->initialized = true;

$ical = new Horde_iCalendar('1.0');
$cal = $object->toiCalendar($ical);
$ical->addComponent($cal);
echo $ical->exportvCalendar() . "\n";

$ical = new Horde_iCalendar('2.0');
$cal = $object->toiCalendar($ical);
$ical->addComponent($cal);
echo $ical->exportvCalendar() . "\n";

$object->setPrivate(true);
$object->setStatus(KRONOLITH_STATUS_TENTATIVE);
$object->recurrence = new Horde_Date_Recurrence($object->start);
$object->recurrence->setRecurType(HORDE_DATE_RECUR_MONTHLY_DATE);
$object->recurrence->setRecurInterval(1);
$object->recurrence->addException(2007, 4, 15);
$object->setAttendees(
    array('juergen@example.com' =>
          array('attendance' => KRONOLITH_PART_REQUIRED,
                'response' => KRONOLITH_RESPONSE_NONE,
                'name' => 'Jürgen Doe'),
          0 =>
          array('attendance' => KRONOLITH_PART_OPTIONAL,
                'response' => KRONOLITH_RESPONSE_ACCEPTED,
                'name' => 'Jane Doe'),
          'jack@example.com' =>
          array('attendance' => KRONOLITH_PART_NONE,
                'response' => KRONOLITH_RESPONSE_DECLINED,
                'name' => 'Jack Doe'),
          'jenny@example.com' =>
          array('attendance' => KRONOLITH_PART_NONE,
                'response' => KRONOLITH_RESPONSE_TENTATIVE)));

$ical = new Horde_iCalendar('1.0');
$cal = $object->toiCalendar($ical);
$ical->addComponent($cal);
echo $ical->exportvCalendar() . "\n";

$ical = new Horde_iCalendar('2.0');
$cal = $object->toiCalendar($ical);
$ical->addComponent($cal);
echo $ical->exportvCalendar() . "\n";

?>
