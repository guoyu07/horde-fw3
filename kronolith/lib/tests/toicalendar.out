Warning: include_once(/home/wrobel/usr/devel/gh/horde-glue/horde-fw3/kronolith/lib/Driver/.php): failed to open stream: No such file or directory in /home/wrobel/usr/devel/gh/horde-glue/horde-fw3/kronolith/lib/Driver.php on line 198

Warning: include_once(): Failed opening '/home/wrobel/usr/devel/gh/horde-glue/horde-fw3/kronolith/lib/Driver/.php' for inclusion (include_path='.:/usr/share/php5:/usr/share/php') in /home/wrobel/usr/devel/gh/horde-glue/horde-fw3/kronolith/lib/Driver.php on line 198

A fatal error has occurred

You must configure a DataTree backend to use Shares.

Details have been logged for the administrator.